### PR TITLE
Fix canary tests v0.3.0

### PR DESCRIPTION
**Pull request by akartsky**
1] Cherry-pick some of the fixes from main
2] For all notebook tests retry failed tests once

### CODE REVIEW

File: test/e2e/tests/test_notebook_instance.py [hunk 1]
```diff
@@ -84,6 +86,7 @@ def get_notebook_instance_resource_status(reference: k8s.CustomResourceReference
     return resource["status"]["notebookInstanceStatus"]
 
 
+@flaky
```
Comment (surajkota) [inline test/e2e/tests/test_notebook_instance.py:89]: Great! how do you set num of retires? is there a default?

**Review by surajkota — APPROVED** [review]
approving assuming the default num retries < 2

File: test/e2e/tests/test_notebook_instance.py (hunk 1 above)
Comment (akartsky) [reply test/e2e/tests/test_notebook_instance.py:89]: we can set it like this `@flaky(max_runs=3, min_passes=2)`

default is `max_runs=2, min_passes=1`

**Review by ryansteakley — APPROVED** [review]

**Pull request merged**